javagen/src/main/java/com/azure/autorest/Javagen.java

```diff
@@ -172,6 +173,26 @@ protected NodeTuple representJavaBeanProperty(Object javaBean, Property property
                 javaPackage.addModuleInfo(client.getModuleInfo());
             }
 
+            // CHANGELOG.md
+            if (settings.isLowLevelClient() && settings.isLlcGenerateNonCode()) {
```

Comment:
I think it's a good idea to remove the low level client constraint - HLCs can take advantage of generating change logs, poms, readmes too. It's just a suggestion - feel free to start with supporting LLCs only.